xarray/core/resample.py

```diff
@@ -80,12 +106,17 @@ def _upsample(self, method, *args, **kwargs):
                 self._obj = self._obj.drop_vars(k)
 
         if method == "asfreq":
-            return self.mean(self._dim)
+            return self.mean(self._dim)  # type: ignore[attr-defined]
```

Comment:
Yeah, I did not want to touch this since I was scared of breaking things exactly for this reason.
Maybe better to leave the ignore for now and deal with that once the reductions are added to generate_reductions (https://github.com/pydata/xarray/issues/6703)?

Reply:
Using `-> T_Xarray` for `_upsample` and `{"asfreq", "pad", "ffill", "backfill", "bfill", "nearest", "interpolate"}` isn't specific enough for `DataArrayResample` right?

I wonder if we should try to move those methods to `generate_reductions` and `DataArrayResampleReductions`? It's the only way I can think of that doesn't seem super hacky at the moment.

Is it difficult to extend the resample reduction with a few methods like this in `generate_reductions`?

Reply:
> resample and groupby reductions are already in `generate_reductions`... @headtr1ck What are you referring to?

sorry, I was mixing something up. You are right, they are added.

Anyway, I could not find a quick way of solving this issue since `Resample` is way before `DataArrayResampleReductions` in the mro, so one cannot simply add an abstract mean function there.
Any ideas?